CommandDotNet.Tests/HelpTests/UsageTests.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using CommandDotNet.Attributes;
+using FluentAssertions;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace CommandDotNet.Tests.HelpTests
+{
+    public class UsageTests : TestBase
+    {
+        private List<string> _output = new List<string>();
+        
+        public UsageTests(ITestOutputHelper testOutputHelper) : base(testOutputHelper)
+        {
+            base.CaptureOutputLine(CaptureOutput);
+        }
+
+        private void CaptureOutput(object sender, string e)
+        {
+            _output.Add(e);
+        }
+
+        [Fact]
+        public void UsageShouldPrintSubcommandsInCorrectOrder()
+        {
+            new AppRunner<HelpApp>().Run("SubHelpCommand", "Print", "--help");
+            var helpText = _output.First().Split(new[] {"\r\n", "\r", "\n"}, StringSplitOptions.RemoveEmptyEntries);
+            var usage = helpText.Single(o => o.StartsWith("Usage")).Substring("Usage: ".Length);
```

Comment:
ummm., this looks too fragile.

Reply:
I agree to an extent.  Parsing strings is generally fragile, and so is comparing on strings.  The well-defined format can let us create a parser to assert subsections of the output.  If the format changes, the parser can be changed to fix all tests.  If we compare on the full output, every test will have to be updated if the format changes.